packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts

```diff
@@ -159,14 +159,35 @@ describe('User Pool', () => {
     expect(() => new UserPool(stack, 'Pool5', {
       userVerification: {
         emailStyle: VerificationEmailStyle.LINK,
-        emailBody: 'invalid email body {####}',
+        emailBody: 'valid email body {####}',
       },
-    })).toThrow(/Verification email body/);
+    })).not.toThrow();
 
     expect(() => new UserPool(stack, 'Pool6', {
       userVerification: {
         emailStyle: VerificationEmailStyle.LINK,
-        emailBody: 'invalid email body {##Verify Email##}',
+        emailBody: 'valid email body {##Verify Email##}',
+      },
+    })).not.toThrow();
+
+    expect(() => new UserPool(stack, 'Pool7', {
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body ##Verify Email##',
+      },
+    })).toThrow(/Verification email body/);
+
+    expect(() => new UserPool(stack, 'Pool8', {
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body {##Verify !! Email##}',
```

Comment:
from the [docs](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-email-verification-message-customization.html), it says that `The maximum length for the message, including the verification code (if present), is 20,000 UTF-8 characters. You can use HTML tags in this message to format the contents.`. so I believe some special characters (like `!`) are allowed? if so we might want to update the regex accordingly

Reply:
good catch, looking at the [cfn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-verificationmessagetemplate.html#cfn-cognito-userpool-verificationmessagetemplate-emailmessagebylink), the regex they used is `\{##[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*##\}` which includes symbol, punctuation and accent characters. Let me update it.